src/type-hierarchy.ts

```diff
@@ -116,8 +118,16 @@ class TypeHierarchyFeature implements vscodelc.StaticFeature {
   initialize(capabilities: vscodelc.ServerCapabilities,
              documentSelector: vscodelc.DocumentSelector|undefined) {
     const serverCapabilities: vscodelc.ServerCapabilities&
-        {typeHierarchyProvider?: any} = capabilities;
+        {standardTypeHierarchyProvider?: any, typeHierarchyProvider?: any} =
+        capabilities;
+    // If this is a clangd that supports standard protocol, don't enable the
+    // extension.
+    if (serverCapabilities.standardTypeHierarchyProvider) {
```

Comment:
Nit: the comment doesn't actually mention the unusual circumstance that leads us to do this weird dance.
Without that it's pretty hard to understand the cases. I also think the control flow here (mixing early return with a second condition) is a bit hard to follow. Maybe:

```
// Unfortunately clangd used the same capability name for its pre-standard protocol
// as the standard ended up using. We need to prevent vscode-languageclient from
// trying to query clangd versions that speak the incompatible protocol.
if (typeHierarchyProvider && !standardTypeHierarchyProvider) {
  // clangd speaks pre-standard protocol. Disable library support for the standard protocol.
  ...
} else {
  // Either we're speaking the standard protocol, or there's no implementation at all.
  // In any case, there's nothing to do.
}
```

```